torch_xla/distributed/parallel_loader.py

```diff
@@ -36,11 +37,16 @@ def __len__(self):
     return self._loader.per_device_samples()
 
   def next(self):
-    if self._mark_step_batch_count == self._batches_yielded:
-      self._batches_yielded = 0
-      xm.mark_step()
-    else:
+    if xp.get_tracer_marked_step():
+      xp.set_tracer_marked_step(False)
       self._batches_yielded += 1
+    else:
+      if self._mark_step_batch_count == self._batches_yielded:
```

Comment:
If profiler is marking step (say, for 100 iters), `self._batches_yielded` might grow too large and then we may never end up marking step afterwards, w/ the `==` check. Let's change to `<=`.